tests/integration/test_vectordraw.py

```diff
@@ -297,25 +302,23 @@ def line_has_desc(self, line):
         aria_describedby = line.get_attribute("aria-describedby")
         return aria_describedby == "jxgboard1-vector-properties"
 
-    def find_line(self, position, line_elements):
-        expected_line_position = list(position.items())
+    def find_line(self, expected_line_position, line_elements):
         for line in line_elements:
-            line_position = list({
+            line_position = {
                 "x1": int(line.get_attribute("x1").split(".", 1)[0]),
                 "y1": int(line.get_attribute("y1").split(".", 1)[0]),
                 "x2": int(line.get_attribute("x2").split(".", 1)[0]),
                 "y2": int(line.get_attribute("y2").split(".", 1)[0]),
-            }.items())
+            }
```

Comment:
THIS was the culprit! For some reason we've been comparing a list of dictionary items instead of a dictionary. Since order isn't guarenteed, this is causing the flakiness (sometimes it isn't in x1, y1, x2, x2 order).

I dug into the commit history to try to discovery if there was a logical/historical reason for converting a dictionary to dictionary items, and then to a list. Turns out it was originally implemented as comparing a *set* of dict items, which was logically correct. [This review comment](https://github.com/open-craft/xblock-vectordraw/commit/deac8d0a288f9dc5f8bff4f6dc789bab8960b7c8#r14385889) suggested directly comparing the dictionaries, instead of converting to a set of tuples (dict items), which would have had the same effect without the extra code.  Unfortunately, when this review comment was addressed ([here](https://github.com/open-craft/xblock-vectordraw/pull/1/commits/a67ccc114bce2d66157d33eb9f2a3efed34df152#diff-78493993c7a04369040380e6e584e09bL194-R193)), the conversion to a set was removed but not `.items()` method call, resulting in a comparison that relied on the order of dictionary items to remain consistent. And here we see a small oversight or typo produce a heisenbug that lay dormant for over 4 years. 

CC @itsjeyd

Reply:
Same goes for the comparison in the `find_point` method below.

Reply:
@swalladge Nice find! :)